amp stats panic

Sometimes, the command `amp stats` causes a panic and crashes amplifier.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xc0 pc=0x544af5]

goroutine 583 [running]:
panic(0xc1c960, 0xc42000c060)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/appcelerator/amp/api/rpc/stats.StatsReply.Less(0xc42027e280, 0xf, 0xf, 0xe, 0x0, 0x0)
        /go/src/github.com/appcelerator/amp/api/rpc/stats/stats.go:452 +0x45
github.com/appcelerator/amp/api/rpc/stats.(*StatsReply).Less(0xc420696c20, 0xe, 0x0, 0x0)
        <autogenerated>:3 +0x78
sort.medianOfThree(0x13de0c0, 0xc420696c20, 0x0, 0x7, 0xe)
        /usr/local/go/src/sort/sort.go:78 +0x7e
sort.doPivot(0x13de0c0, 0xc420696c20, 0x0, 0xf, 0x19, 0x0)
        /usr/local/go/src/sort/sort.go:103 +0xa7
sort.quickSort(0x13de0c0, 0xc420696c20, 0x0, 0xf, 0x7)
        /usr/local/go/src/sort/sort.go:188 +0x83
sort.Sort(0x13de0c0, 0xc420696c20)
        /usr/local/go/src/sort/sort.go:222 +0x80
github.com/appcelerator/amp/api/rpc/stats.(*Stats).StatsQuery(0xc4203681e0, 0x13e2fc0, 0xc4207233b0, 0xc42098aa20, 0x0, 0xb, 0xb)
        /go/src/github.com/appcelerator/amp/api/rpc/stats/stats.go:68 +0x1a2
github.com/appcelerator/amp/api/rpc/stats._Stats_StatsQuery_Handler(0xce9d80, 0xc4203681e0, 0x13e2fc0, 0xc4207233b0, 0xc4205631d0, 0x0, 0x0, 0x0, 0xc42006e358, 0x0)
        /go/src/github.com/appcelerator/amp/api/rpc/stats/stats.pb.go:470 +0x27d
github.com/appcelerator/amp/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc420520780, 0x13e5b60, 0xc4208605a0, 0xc420218ff0, 0xc42050cfc0, 0x13ba710, 0xc420723380, 0x0, 0x0)
        /go/src/github.com/appcelerator/amp/vendor/google.golang.org/grpc/server.go:638 +0xc14
github.com/appcelerator/amp/vendor/google.golang.org/grpc.(*Server).handleStream(0xc420520780, 0x13e5b60, 0xc4208605a0, 0xc420218ff0, 0xc420723380)
        /go/src/github.com/appcelerator/amp/vendor/google.golang.org/grpc/server.go:796 +0x6b0
github.com/appcelerator/amp/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc420a8c7b0, 0xc420520780, 0x13e5b60, 0xc4208605a0, 0xc420218ff0)
        /go/src/github.com/appcelerator/amp/vendor/google.golang.org/grpc/server.go:449 +0xab
created by github.com/appcelerator/amp/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
        /go/src/github.com/appcelerator/amp/vendor/google.golang.org/grpc/server.go:450 +0xa3
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -20,6 +20,7 @@ install:
 
 script:
   - make test
+  - amp logs amplifier
 
 after_success:
   - if [ "$TRAVIS_BRANCH" == "master" ] && [ "$TRAVIS_PULL_REQUEST" == "false" ]; then

--- a/.travis.yml
+++ b/.travis.yml
@@ -6,7 +6,7 @@ services:
 language: go
 
 go:
-  - 1.7.3
+  - 1.7.4
 
 install:
   - go get -u github.com/golang/lint/golint
@@ -20,7 +20,6 @@ install:
 
 script:
   - make test
-  - amp logs amplifier
 
 after_success:
   - if [ "$TRAVIS_BRANCH" == "master" ] && [ "$TRAVIS_PULL_REQUEST" == "false" ]; then

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,5 +1,5 @@
-# appcelerator/protoc is based on alpine and includes latest go and protoc
-FROM appcelerator/protoc:0.3.0
+# golang:alpine provides an up to date go build environment
+FROM golang:alpine
 RUN echo "@community http://nl.alpinelinux.org/alpine/edge/community" >> /etc/apk/repositories
 RUN apk --no-cache add bash alpine-sdk
 WORKDIR /go/src/github.com/appcelerator/amp

--- a/api/rpc/stats/stats.go
+++ b/api/rpc/stats/stats.go
@@ -65,7 +65,7 @@ func (s *Stats) StatsQuery(ctx context.Context, req *StatsRequest) (*StatsReply,
 		}
 		return ret, nil
 	}
-	sort.Sort(result)
+	sort.Sort(s.cleanUp(result))
 	return result, nil
 }
 
@@ -105,6 +105,7 @@ func (s *Stats) combineStats(req *StatsRequest, list *[4]*StatsReply) *StatsRepl
 			}
 		}
 	}
+
 	return finalRet
 
 }
@@ -159,6 +160,17 @@ func (s *Stats) updateRow(ref *StatsEntry, row *StatsEntry) {
 		ref.NetTxBytes = row.NetTxBytes
 		ref.NetRxBytes = row.NetRxBytes
 	}
+}
+
+func (s *Stats) cleanUp(result *StatsReply) *StatsReply {
+	ret := &StatsReply{}
+	for _, row := range result.Entries {
+		if row != nil {
+			ret.Entries = append(ret.Entries, row)
+		}
+	}
+	return ret
+
 }
 
 // statsQueryMetric extracts stat information according to StatsRequest for one  metric (cpu | mem | io | net)

--- a/api/rpc/stats/stats.go
+++ b/api/rpc/stats/stats.go
@@ -65,8 +65,9 @@ func (s *Stats) StatsQuery(ctx context.Context, req *StatsRequest) (*StatsReply,
 		}
 		return ret, nil
 	}
-	sort.Sort(s.cleanUp(result))
-	return result, nil
+	clean := s.cleanUp(result)
+	sort.Sort(clean)
+	return clean, nil
 }
 
 func (s *Stats) addStatsResult(list *[4]*StatsReply, ret *StatsReply) {

